### PR TITLE
Add users

### PR DESCRIPTION
**Pull request by herrone**
added for this week

### CODE REVIEW

File: Utils/PasswordHasher.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using System.Security.Cryptography;
+using System.Text;
+
+public static class PasswordHasher
+{
+    public static string HashPassword(string password)
+    {
+        using (var sha256 = SHA256.Create())
+        {
+            var hashedBytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(password));
```
Comment (Pannoniae) [inline Utils/PasswordHasher.cs:11]: good job on password hashing!

File: Views/UsersPage.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        {
+            string email = txtLoginEmail.Text; string password = txtLoginPassword.Text;
+            var user = users.FirstOrDefault(u => u.Email == email && PasswordHasher.VerifyPassword(password, u.Password));
+
+            if (user != null)
+            {
+                if (user.Role == "User")
+                {
+                    await DisplayAlert("Notice", "Contact manager", "OK");
+                }
```
Comment (Pannoniae) [inline Views/UsersPage.xaml.cs:48]: using else if's would greatly reduce the nesting, which would make it way more readable

File: Views/UsersPage.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+            // For example: loginSection.IsVisible = false;
+        }
+
+
+        private async void AddUserButton_Clicked(object sender, EventArgs e)
+        {
+            if (txtName.Text != null)
+            {
+                if (txtEmail.Text != null)
+                {
```
Comment (Pannoniae) [inline Views/UsersPage.xaml.cs:93]: same here, inverting the conditions would reduce nesting greatly.

would also help to refactor this into multiple methods

**Review by Pannoniae — APPROVED** [review]
Nice PR!

File: Views/UsersPage.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+                if (email == "1" && password == "1")
+                {
+                    userManagementSection.IsVisible = true;
+                }
+                else
+                {
+                    await DisplayAlert("Login Failed", "Invalid credentials", "OK");
+                }
+            }
+        }
```
Comment (stefan4h) [inline Views/UsersPage.xaml.cs:75]: The code contains multiple nested if statements, making it difficult to read and understand. It could be simplified significantly by using guard clauses. Also, the code's formatting is inconsistent (e.g., the first two variable declarations are on the same line), and the method checks for the role and email in a way that could be simplified.

File: Views/UsersPage.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+                            else await DisplayAlert("Error", "Invalid Email", "OK");
+                        }
+                        else await DisplayAlert("Error", "Team is empty", "OK");
+                    }
+                    else await DisplayAlert("Error", "Password is empty", "OK");
+                }
+                else await DisplayAlert("Error", "Email is empty", "OK");
+            }
+            else await DisplayAlert("Error", "Name is empty", "OK");
+        }
```
Comment (stefan4h) [inline Views/UsersPage.xaml.cs:131]: the code has too many nested if statements, making it hard to follow, there are also unnecessary checks for null on text fields that are typically empty strings rather than null in UI contexts, and there's no need for multiple await DisplayAlert statements for each field check, they just lead to repetition and clutter. Try implementing this code using guard clauses

**Review by stefan4h — CHANGES_REQUESTED** [review]
